sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -590,6 +590,14 @@ class InMemoryCatalog(
     catalog(db).functions.remove(funcName)
   }
 
+  override protected def doAlterFunction(db: String, func: CatalogFunction): Unit = synchronized {
+    requireDbExists(db)
+    requireFunctionExists(db, func.identifier.funcName)
+    catalog(db).functions.remove(func.identifier.funcName)
+    requireFunctionNotExists(db, func.identifier.funcName)
```

Comment:
It looks weird you remove a function and then require it not exists, are you worried about removing function may be wrong?